docs/api/fetchBaseQuery.md

```diff
@@ -0,0 +1,147 @@
+        query: ({ name, patch }) => ({
+          url: `pokemon/${name}`,
+          method: 'PATCH', // When making a mutation, we specify the method of PATCH/PUT/POST for REST endpoints
+          body: patch, // fetchBaseQuery automatically adds `content-type: application/json` and calls JSON.stringify(patch) for you
+        })
+      },
+    })
+  }),
+});
+```
+
+### Individual query options
+
+Even though `fetchBaseQuery` only takes an object with `baseUrl`, there is more behavior that you can define on a per-request basis.
+
+```ts
+interface FetchArgs extends RequestInit {
+  url: string;
+  params?: Record<string, any>;
+  body?: any;
+  responseHandler?: 'json' | 'text' | ((response: Response) => Promise<any>);
+  validateStatus?: (response: Response, body: any) => boolean;
+}
+
+const defaultValidateStatus = (response: Response) => response.status >= 200 && response.status <= 299;
+```
+
+### Setting the body
+
+By default, we assume that every request you make will be `json`, so in those cases all you have to do is set the `url` and pass a `body` object. For other implementations, you can manually set the `Headers` to specify the content type.
+
+#### json
+
+```ts
+ // omitted
+  endpoints: (builder) => ({
+    updateUser: builder.query({
+      query: (user: Record<string, string>) => ({
+        url: `users`,
+        method: 'PUT',
+        body: user // Body is automatically converted to json with the correct headers
+      }),
+    }),
+```
+
+#### text
+
+```ts
+ // omitted
+  endpoints: (builder) => ({
+    updateUser: builder.query({
+      query: (user: Record<string, string>) => ({
+        url: `users`,
+        method: 'PUT',
+        headers: {
+            'content-type': 'text/plain',
+        },
+        body: user
+      }),
+    }),
+```
+
+### Setting the query string
+
+`fetchBaseQuery` provides a simple wrapper like you'd find elsewhere that converts an `object` to serialized query params for you.
+
+```ts
+  endpoints: (builder) => ({
+    updateUser: builder.query({
+      query: (user: Record<string, string>) => ({
+        url: `users`,
+        params: user // The user object is automatically converted and produces a request like /api/users/?first_name=test&last_name=example
+      }),
+    }),
+```
+
+### Parsing a Response
+
+By default, we'll assume that every `Response` you get will be parsed as `json`. In the event that you don't want that to happen, you can specify an alternative response handler like `text`, or take complete control and use a custom function that takes the raw `Response` object &mdash; allowing you to use any [`Body` method](https://developer.mozilla.org/en-US/docs/Web/API/Body).
+
+```ts title="Parse a response as formData"
+export const customApi = createApi({
+  baseQuery: fetchBaseQuery({ baseUrl: '/api/' }),
+  endpoints: (builder) => ({
+    getUsers: builder.query({
+      query: () => ({
+        url: `users`,
+        responseHandler: (response) => response.formData(),
```

Comment:
this would lead to a non-serializable value in the resolved action, so we should probably pick a different example (if there are other things as text & json?)

Reply:
Yea, that's definitely something to consider. I'm assuming we should add a note about serialized values and store configuration? Could use some ideas on the right approach there.